lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -711,7 +848,8 @@ SILCombiner::createApplyWithConcreteType(FullApplySite Apply,
   }
   // The apply can only be rewritten in terms of the concrete value if it is
   // legal to pass that value as the self argument.
-  if (CEI.isCopied && !canReplaceCopiedSelf(Apply, CEI.InitExistential, DA))
+  if (CEI.InitExistential && CEI.isCopied &&
+      !canReplaceCopiedSelf(Apply, CEI.InitExistential, DA))
```

Comment:
Is this correct?

Reply:
@atrick suggested this: if (isCopied && (!InitExistential || !canReplaceCopiedSelf(Apply, InitExistential)
which seem reasonable.